src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -1905,6 +2206,49 @@ public ManifestPublishingCallbackAnnotation(System.Func<Publishing.ManifestPubli
         public static ManifestPublishingCallbackAnnotation Ignore { get { throw null; } }
     }
 
+    public partial record NetworkEndpointSnapshot(ValueSnapshot<AllocatedEndpoint> Snapshot, NetworkIdentifier NetworkID)
+    {
+    }
+
+    public partial class NetworkEndpointSnapshotList : System.Collections.Generic.IEnumerable<NetworkEndpointSnapshot>, System.Collections.IEnumerable
+    {
+        public System.Collections.Generic.IEnumerator<NetworkEndpointSnapshot> GetEnumerator() { throw null; }
+
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+
+        public bool TryAdd(NetworkIdentifier networkID, ValueSnapshot<AllocatedEndpoint> snapshot) { throw null; }
+    }
+
+    public readonly partial struct NetworkIdentifier : System.IEquatable<NetworkIdentifier>
```

Comment:
Does this need to be a `record struct`?

Reply:
OK so make it a `class` with public fields? (just `Value` or `Id` for starters)?

Reply:
Do we know we want to eventually evolve it to a compound identitier? If so, yeah maybe making it a `class` or `record` would be better.

Reply:
I am not certain about it but it would be really unfortunate if we closed this possibility now. I am leaning towards `record`